lib/bindings/ARGOBinding.js

```diff
@@ -0,0 +1,547 @@
+            // callback(null, device);
+            // })();
+            // },
+
+            // Perfom the entity update
+            function (device, callback) {
+                const attrs = Object.keys(req.body).map((key) => ({
+                    name: key,
+                    type: guessType(key, device, null),
+                    value: req.body[key]
+                }));
+                if (!attrs || attrs.length === 0) {
+                    return callback();
+                }
+                return iotAgentLib.update(device.name, device.type, '', attrs, device, function (error) {
+                    if (error) {
+                        logger.error(req.logContext, 'Error updating device %s: %j', req.deviceId, error);
+                    }
+                    return callback(error);
+                });
+            }
+        ],
+        function (error) {
+            // We decide NOT to generate a new transaction (domain) at this
+            // point. domain API is deprecated and it has been proven to be
+            // very detrimental for performance, the way it is used here.
+            // iotAgentLib.finishSouthBoundTransaction(function () {
+            // next(error);
+            // });
+            next(error);
+        }
+    );
+}
+
+/**
+ * Default error handler for express
+ */
+function handleError(localConf) {
+    return function (error, req, res, next) {
+        if (res.headersSent) {
+            return next(error);
+        }
+        return writeSoapEnvelope(localConf, res, error);
+    };
+}
+
+/**
+ * Send reply
+ */
+function argoReply(localConf) {
+    return function (req, res, next) {
+        return writeSoapEnvelope(localConf, res, null);
+    };
+}
+
+/**
+ * Write ARGO envelope as result
+ */
+function writeSoapEnvelope(localConf, res, error) {
+    let message = 'OK';
+    let status = 200;
+    let returnCode = 0;
+    if (error) {
+        message = error.message || error;
+        status = error.code && String(error.code).match(/^[2345]\d\d$/) ? error.code : 500;
+        returnCode = status;
+    }
+    res.setHeader('content-type', `application/soap+xml; action="urn:${localConf.argoAction}"; charset=utf-8`);
+    return res.status(status).send(`<?xml version="1.0" encoding="UTF-8"?>
+    <soap:Envelope xmlns:soap="http://www.w3.org/2003/05/soap-envelope" xmlns:ser="${localConf.argoNamespace}">
+        <soap:Header/>
+        <soap:Body>
+            <ser:${localConf.argoAction}>
+                <ser:returnCode>${returnCode}</ser:returnCode>
+                <ser:message>${message}</ser:message>
+                <ser:IDCasoExterno/>
+            </ser:${localConf.argoAction}>
+        </soap:Body>
+    </soap:Envelope>`);
+}
+
+/**
+ * Start the ARGO binding
+ */
+function start(callback) {
+    const logger = configService.getLogger();
+    const upstreamConf = configService.getConfig();
+    const localConf = processEnvironmentVariables(logger, {
```

Comment:
Maybe some configuration (host, port, remote_url and a disable flag) could be added to https://github.com/telefonicaid/iotagent-json/blob/master/lib/configService.js

Reply:
the host, port and disable flag do exist, but I kept them all inside this file to make the service self-contained and "pluggable". In the future you could deprecate this feature from the iota-json anytime (e.g. because it migrates to a dedicated iota), just by removing this file and its tests.

But if you feel it is better to move some of the constants, flags or errors to the shared files, I am ok with that too.

Reply:
NTC at the end.